### DISCUSSION TITLE
Running docker images fails

### DISCUSSION THREAD

**Discussion by TIS-Arne** [OE4T/meta-tegra#758] [Q&A] [answered]
I am struggling to get docker working on images build using the tegra-demo-distro. 
Trying to run docker images always results in this error:
```docker: Error response from daemon: failed to create shim: OCI runtime create failed: container_linux.go:380: starting container process caused: process_linux.go:545: container init caused: Running hook #0:: error running hook: exit status 1, stdout: , stderr: nvidia-container-cli: initialization error: driver error: failed to process request: unknown.```

Is there anything more I need to do to run docker images?

**Comment by madisongh** [comment]
Which branch are you using? I've seen this on master in the last couple of days, and haven't had a chance to track down the problem yet.

**Reply by ichergui** [reply]
Hey @TIS-Arne 
Please provide the elements that @madisongh asked about ? Are you using [Nvidia NGC Containers](https://ngc.nvidia.com/catalog/containers/)  ?

I'm asking because I was using docker containers with `dunfell-l4t-r32.5.0` and It works fine to me.

**Comment by TIS-Arne** [comment]
I am on master and yes, I am using the NVIDIA NGC Containers. 
I will switch to dunfell-l4t-r32.5.0.

**Reply by ichergui** [reply]
Yes, please try with `dunfell-l4t-r32.5.0`
I was using the [L4T ML](https://ngc.nvidia.com/catalog/containers/nvidia:l4t-ml) container.
Let us know if you have issues

**Reply by madisongh** [reply]
Yeah, something's wrong on master.   I've opened #760 to track.

**Comment by madisongh** [comment, marked as answer]
I root-caused this and implemented a workaround to deal with the limitations (and, I think bug) in `libtirpc`.  See #760 for more info.

**Reply by ichergui** [reply]
I confirm that it is working perfectly in `master` branch 
Thanks @madisongh 
Test was done on the `jetson-tx2-devkit`. I was using the [L4T ML](https://ngc.nvidia.com/catalog/containers/nvidia:l4t-ml) container.

* Running the container

1.  Pull the l4t-ml container

```
root@jetson-tx2-devkit:~# docker pull nvcr.io/nvidia/l4t-ml:r32.5.0-py3
r32.5.0-py3: Pulling from nvidia/l4t-ml
5000a6c32c5a: Pull complete 
8e855b69096a: Pull complete 
8db8dbbd4bb9: Pull complete 
833dc3235950: Pull complete 
f79d264135a3: Pull complete 
1c40f77bb35b: Pull complete 
1990ecf0bfb7: Pull complete 
c8ffbfd7f0aa: Pull complete 
ba785779122a: Pull complete 
024ce79b6790: Pull complete 
9b09da3b5483: Pull complete 
17f974a43cf9: Pull complete 
211b56b73ff1: Pull complete 
78aca4be1f3b: Pull complete 
95f34310bbda: Pull complete 
678c9d1557e9: Pull complete 
ec17ad7cab01: Pull complete 
08fb1eee5328: Pull complete 
df804e245232: Pull complete 
4f9a01a0e955: Pull complete 
a44515425a95: Pull complete 
cf055aaf10ad: Pull complete 
844538a014b9: Pull complete 
159036d408be: Pull complete 
ae321b977e2f: Pull complete 
ae4d2ced71ce: Pull complete 
1bc11ebf8522: Pull complete 
ef6ff4fc67ed: Pull complete 
3b45df8b8d80: Pull complete 
a1a094c74107: Pull complete 
7400833e9fab: Pull complete 
21608f9f63e2: Pull complete 
50b80108db64: Pull complete 
800ad6d6d249: Pull complete 
a5297af96097: Pull complete 
5d3a862a0013: Pull complete 
418524dee780: Pull complete 
Digest: sha256:af8155d948946e76fb398a93a726ee4e4f06a69574284a8bc8929f0731efdc61
Status: Downloaded newer image for nvcr.io/nvidia/l4t-ml:r32.5.0-py3
nvcr.io/nvidia/l4t-ml:r32.5.0-py3
root@jetson-tx2-devkit:~#
```

2.  Run the container

```
root@jetson-tx2-devkit:~# docker run -it --rm --runtime nvidia --network host nvcr.io/nvidia/l4t-ml:r32.5.0-py3
allow 10 sec for JupyterLab to start @ http://localhost:8888 (password nvidia)
JupterLab logging location:  /var/log/jupyter.log  (inside the container)
root@jetson-tx2-devkit:/#
```

**Comment by TIS-Arne** [comment]
Sorry, it took a while to rebuild and test. Indeed it works with the dunfell branch and I will stick to that one for now. 

The reason to use docker for me was because we need to get a python DeepStream app to work on the OE image. The pyds bindings are for python 3.6 only which is not available for OE. So I tried to use the docker container for DeepStream but this also seems not to be a good solution since I did not find an easy way to build a new container based on the NVIDIA container with all the required dependencies and then integrate that into the image. Has anybody solved this issue already?

I also tried to build Python3.6 and install it in addition to the version that OE uses. This works for the pyds bindings but now I am struggling to get the pygobject bindings to be built for this python3.6 version and without it, using deepstream is also not possible :(

**Reply by madisongh** [reply]
I think the scripts for building the NGC containers are [here](https://github.com/dusty-nv/jetson-containers).

We had a discussion about the Python bindings problem on #425 ... I don't see a good solution to that problem unless NV publishes the sources for the bindings.
